test/no-typeof-undefined.mjs

```diff
@@ -0,0 +1,17 @@
+import {getTester} from './utils/test.mjs';
+
+const {test} = getTester(import.meta);
+
+test.snapshot({
+	valid: [
+		'typeof notDeclared === "undefined"',
+	],
+	invalid: [
+		'var withVar; typeof withVar == "undefined"',
+		'let withLet; "undefined" != typeof withLet',
+		'const withConst = 0; typeof withConst === "undefined"',
+		'import withImport from "foo"; "undefined" !== typeof withImport',
+		'withArrowFnParameter => typeof withArrowFnParameter === "undefined"',
+		'function foo(withFnParameter) { "undefined" !== typeof withFnParameter }',
+	],
```

Comment:
This is gonna need a lot more valid/invalid test cases.

Reply:
added new valid/invalid tests, you can share your code snippets also